cloudsync/api.py

```diff
@@ -74,3 +81,128 @@ def refresh_status(video, encode_job=None):
             log.error('Transcoding failed for video %d', video.id)
         encode_job.message = et_job
         encode_job.save()
+
+
+def transcode_video(video, video_file):
+    """
+    Start a transcode job for a video
+
+    Args:
+        video(ui.models.Video): the video to transcode
+        video_file(ui.models.Videofile): the s3 file to use for transcoding
+
+    """
+
+    video_input = {
+        'Key': video_file.s3_object_key,
+    }
+
+    # Generate an output video file for each encoding (assumed to be HLS)
+    outputs = [{
+        'Key': video.transcode_key(preset),
+        'PresetId': preset,
+        'SegmentDuration': '10.0'
+    } for preset in settings.ET_PRESET_IDS]
+
+    playlists = [{
+        'Format': 'HLSv3',
+        'Name': video.transcode_key('_index'),
+        'OutputKeys': [output['Key'] for output in outputs]
+    }]
+
+    # Generate thumbnails for the 1st encoding (no point in doing so for each).
+    outputs[0]['ThumbnailPattern'] = THUMBNAIL_PATTERN.format(video_file.s3_basename)
+    transcoder = VideoTranscoder(
+        settings.ET_PIPELINE_ID,
+        settings.AWS_REGION,
+        settings.AWS_ACCESS_KEY_ID,
+        settings.AWS_SECRET_ACCESS_KEY
+    )
+    try:
+        transcoder.encode(video_input, outputs, Playlists=playlists)
+    except ClientError as exc:
+        log.error('Transcode job creation failed for video %s', video.id)
+        video.update_status(VideoStatus.TRANSCODE_FAILED)
+        if hasattr(exc, 'response'):
+            transcoder.message = exc.response
+        raise
+    finally:
+        transcoder.create_job_for_object(video)
+        if video.status != VideoStatus.TRANSCODE_FAILED:
+            video.update_status(VideoStatus.TRANSCODING)
```

Comment:
A few lines up from here, if a ClientError occurs when creating the transcode job.  Also later, when transcode job statuses are checked by the periodic celery task.